ddtrace/tracer/tracer.go

```diff
@@ -198,7 +188,13 @@ func newTracer(opts ...StartOption) *tracer {
 	t.wg.Add(1)
 	go func() {
 		defer t.wg.Done()
-		t.worker()
+		tick := t.config.tickChan
+		if tick == nil {
+			ticker := time.NewTicker(flushInterval)
+			defer ticker.Stop()
+			tick = ticker.C
+		}
```

Comment:
> This check seems slightly out of place here.

It's also not far from the truth to say that this is exactly in the right place because it's where it is used. Also, consider that if it were to be where you are suggesting, imagine how the logic involved with stopping the ticker would look like... You'd have to save the ticker somewhere (a field?) and then stop it here. This makes it compact and obvious how it works. I feel logic would become fragmented otherwise.

I do see your point, but since this is technically a test hook, and adding it with the rest of the defaults would complicate things with stopping the ticker, I feel this is much better.

> Also, are we leaking a ticker by not calling ticker.Stop() if t.config.tickChan != nil? I don't see it stopped in the tests.

`tick` is a channel, not a ticker. We don't need to stop it.